src/careamics/dataset/tiling/lvae_tiled_patching.py

```diff
@@ -93,19 +96,98 @@ def extract_tiles(
             yield tile, tile_info
 
 
+def compute_tile_info_legacy(
+    grid_index_manager: GridIndexManager, index: int
+) -> TileInformation:
+    """
+    Compute the tile information for a tile at a given dataset index.
+
+    Parameters
+    ----------
+    grid_index_manager : GridIndexManager
+        The grid index manager that keeps track of tile locations.
+    index : int
+        The dataset index.
+
+    Returns
+    -------
+    TileInformation
+        Information that describes how to crop and stitch a tile to create a full image.
+
+    Raises
+    ------
+    ValueError
+        If `grid_index_manager.data_shape` does not have 4 or 5 dimensions.
+    """
+    data_shape = np.array(grid_index_manager.data_shape)
```

Comment:
I think this might just work. however, it is worth noting that this can be a potential issue. Let me explain
1. in some cases, where we just have the input channel, data_shape will have 1 as the channel dimension but we will be predicting >2 channels. So it made sense to pick the channel dimension from the prediction. In this case, where there is no prediction, one could pass the information about how many channels are getting predicted.

Reply:
I have already encountered this issue, and I hacked a fix into the careamics stitch prediction function [here](https://github.com/CAREamics/careamics/blob/427fa268a4b6a0cd1db445834bf440e541dda1f3/src/careamics/prediction_utils/stitch_prediction.py#L79-L94), in a previous PR. It's not very elegant but careamics might have to deal with this in other applications so I will leave it like this for now.